protocol/x/accountplus/authenticator/message_filter.go

```diff
@@ -0,0 +1,82 @@
+package authenticator
```

Comment:
Osmosis's implementation of `message_filter` actually matches full JSON patterns ([link](https://github.com/osmosis-labs/osmosis/blob/main/x/smart-account/authenticator/message_filter.go#L79)), which is an overkill for our usecase (might have some perf implications as well). 

For our implementation, I rewrote `message_filter` to only filter on MsgTypeURL for simplicity.